File: back-end/src/main/java/com/huyphan/dtos/BookPaginationOptionsDto.java
package com.huyphan.dtos;

import java.util.List;
import com.huyphan.models.enums.BookOrderField;
import com.huyphan.models.enums.OrderDirection;
import lombok.Getter;
import lombok.Setter;

/** Book pagination options DTO. */
@Getter
@Setter
public class BookPaginationOptionsDto extends PaginationOptionsDto {

	/** Order direction. */
	private OrderDirection orderDirection;

	/** Order field. */
	private BookOrderField orderField;

	/** Search criteria list. */
	private List<SearchCriteriaDto> criteria;
}

File: front-end/src/app/core/services/dtos/book-pagination-options-dto.ts
import { BookOrderField } from '../../enums/book-order-field';
import { OrderDirection } from '../../enums/order-direction';

import { PaginationOptionsDto } from './pagination-options-dto';
import { SearchCriteriaDto } from './search-criteria-dto';

/** Book pagination options DTO. */
export interface BookPaginationOptionsDto extends PaginationOptionsDto {
  /** Order direction. */
  readonly orderDirection: OrderDirection;

  /** Order field. */
  readonly orderField: BookOrderField;

  /** Search criteria list. */
  readonly criteria: readonly SearchCriteriaDto<unknown>[];
}
